### PR TITLE
Make method static

### LINKED ISSUE #364: Make method static if possible
If method in class is not using any instance fields it can be made static.

````csharp
class Test 
{
    public char GetThirdCharacterFromString(string str)
    {
        return str[2];
    }
}
class Referencing
{
    void Foo()
    {
        var t = new Test();
        t.GetThirdCharacterFromString("");
    }
}
````

becomes

````csharp
public class Test 
{
    public static char GetThirdCharacterFromString(string str)
    {
        return str[2];
    }
    void Foo()
    {
        var t = new Test();
        Test.GetThirdCharacterFromString("");
    }
}
````

A non static method can be turned static, but only if it is not accessing `this`.
Also, the code fix would have to find all references and change them to static invocations. If the method is public, on the whole solution, if it is internal, on the project, etc. The current roslyn infrastructure takes care of finding those references for us.

Diagnostic id: `CC0091`
Category: `Design`
Severity: `Info`

### PR DESCRIPTION
**Pull request by giggio**
closes #364

### CODE REVIEW

File: src/CSharp/CodeCracker/Design/MakeMethodStaticAnalyzer.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context) =>
+            context.RegisterSyntaxNodeAction(LanguageVersion.CSharp6, AnalyzeMethod, SyntaxKind.MethodDeclaration);
+
+        private void AnalyzeMethod(SyntaxNodeAnalysisContext context)
+        {
+            if (context.IsGenerated()) return;
+            var method = (MethodDeclarationSyntax)context.Node;
+            if (method.Modifiers.Any(SyntaxKind.StaticKeyword)) return;
```
Comment (baks) [inline src/CSharp/CodeCracker/Design/MakeMethodStaticAnalyzer.cs]: The analyzer should also immediately return if there are any of this method modifiers:
* `virtual`
* `override`
* `new`

Comment (giggio) [inline src/CSharp/CodeCracker/Design/MakeMethodStaticAnalyzer.cs]: @baks makes sense. I will updated it.

Comment (giggio) [inline src/CSharp/CodeCracker/Design/MakeMethodStaticAnalyzer.cs]: @baks Done. Also added `abstract`.

File: src/CSharp/CodeCracker/Design/MakeMethodStaticAnalyzer.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+namespace CodeCracker.CSharp.Design
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class MakeMethodStaticAnalyzer : DiagnosticAnalyzer
+    {
+        internal const string Title = "Use static method";
+        internal const string MessageFormat = "Make '{0}' method static.";
+        internal const string Category = SupportedCategories.Design;
+        const string Description = "If you are not referencing any instance variable your method may become static. That will require";
```
Comment (baks) [inline src/CSharp/CodeCracker/Design/MakeMethodStaticAnalyzer.cs]: It looks like something is missing in `Description` string or am I wrong? :question:

Comment (giggio) [inline src/CSharp/CodeCracker/Design/MakeMethodStaticAnalyzer.cs]: @baks  I think I forgot to finish it. I updated and rebased everything.

**Comment by carloscds** [comment]
@giggio Giovanni, can you update your PR for merge ?

**Comment by giggio** [comment]
@carloscds done!

**Pull request merged**